public/app/plugins/datasource/azuremonitor/components/ResourcePicker/utils.ts

```diff
@@ -13,7 +13,7 @@ import { ResourceRow, ResourceRowGroup } from './types';
 //  - resource groups: /subscriptions/44693801-6ee6-49de-9b2d-9106972f9572/resourceGroups/cloud-datasources
 //  - resources: /subscriptions/44693801-6ee6-49de-9b2d-9106972f9572/resourceGroups/cloud-datasources/providers/Microsoft.Compute/virtualMachines/GithubTestDataVM
 const RESOURCE_URI_REGEX =
-  /\/subscriptions\/(?<subscription>[^/]+)(?:\/resourceGroups\/(?<resourceGroup>[^/]+)(?:\/providers\/(?<metricNamespaceAndResource>.+))?)?/;
+  /\/subscriptions\/(?<subscription>[^/]+)(?:\/resourceGroups\/(?<resourceGroup>[^/]+)(?:\/providers\/(?<metricNamespaceAndResource>.+))?)?/i;
```

Comment:
added the `i` flag to make this case insensitive. I found that for some cases (for example doing `.replace()` on a workspace variable) was returning everything in lower case so this regex would not properly match all groups because of the casing in `resourceGroups`

Reply:
Nice catch!